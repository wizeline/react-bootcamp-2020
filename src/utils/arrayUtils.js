const { MAX_VIDEOS_TO_FETCH } = require('./constants');

const filterLastRelatedVideos = (array, id) => {
  if (!array ||Â !id) return [];
  const initialIndex = array.length - MAX_VIDEOS_TO_FETCH;
  const finalIndex = array.length;
  const filtered = array
    .slice(initialIndex, finalIndex)
    .filter((video) => video.id.videoId !== id);
  return filtered;
};

const findVideo = (videoList, favoriteId) => {
  if (!videoList || !favoriteId) return false;
  return videoList.find((element) => element.id.videoId === favoriteId);
};

const removeVideoFromFavorites = (videoList, favoriteId) => {
  if (!videoList || !favoriteId) return [];
  return videoList.filter(e => e).filter((element) => element.id.videoId !== favoriteId);
};

const removeRepeatedVideos = (videoList) => {
  if (!videoList) return [];
  return videoList.filter(
    (v, i, a) => a.findIndex((t) => t.id.videoId === v.id.videoId) === i
  );
};

export {
  filterLastRelatedVideos,
  findVideo,
  removeVideoFromFavorites,
  removeRepeatedVideos,
};
