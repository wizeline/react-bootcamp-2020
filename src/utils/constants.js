const AUTH_STORAGE_KEY = 'wa_cert_authenticated';
const MAX_VIDEOS_TO_FETCH = 5;

export { AUTH_STORAGE_KEY, MAX_VIDEOS_TO_FETCH };
