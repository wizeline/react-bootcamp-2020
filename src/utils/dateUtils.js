import moment from 'moment';

const formatDate = (date) => {
  if (!date) return '';
  const formattedDate = moment(date).format('DD/MM/YYYY');
  return formattedDate;
};

export { formatDate };
