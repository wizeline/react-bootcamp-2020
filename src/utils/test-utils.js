// test-utils.js
import React from 'react'
import { render as rtlRender } from '@testing-library/react'
import { createStore } from 'redux'
import { Provider } from 'react-redux'
// Import your own reducer
import reducer from '../redux/reducers/index'

function render(
  ui,
  {
    initialState,
    store = createStore(reducer, initialState),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions })
}

// re-export everything
export * from '@testing-library/react'
// override render method

export const thunk = ({ dispatch, getState }) => next => action => {
  if (typeof action === 'function') {
    return action(dispatch, getState)
  }

  return next(action)
}

export const create = () => {
  const store = {
    getState: jest.fn(() => ({})),
    dispatch: jest.fn()
  }
  const next = jest.fn()

  const invoke = action => thunk(store)(next)(action)

  return {store, next, invoke}
}

export default {
  render,
  thunk,
  store: create().store,
  next: create().invoke,
  invoke: create().store
}