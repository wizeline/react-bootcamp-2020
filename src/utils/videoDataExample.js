const videos = [
  {
    kind: 'youtube#searchResult',
    etag: 'qTtZNL0lYW0zWZCHMtnW1Zu1Sfs',
    id: {
      kind: 'youtube#video',
      videoId: 'SiDYFCd9SH4',
    },
    snippet: {
      publishedAt: '2020-09-26T02:26:02Z',
      channelId: 'UCXIJgqnII2ZOINSWNOGFThA',
      title: 'Hannity slams the left&#39;s attacks on Judge Amy Coney Barrett',
      description:
        'President Trump expected to nominate appeals court judge to the Supreme Court Saturday. #FoxNews #Hannity Subscribe to Fox News! https://bit.ly/2vBUvAS ...',
      thumbnails: {
        default: {
          url: 'https://i.ytimg.com/vi/SiDYFCd9SH4/default.jpg',
          width: 120,
          height: 90,
        },
        medium: {
          url: 'https://i.ytimg.com/vi/SiDYFCd9SH4/mqdefault.jpg',
          width: 320,
          height: 180,
        },
        high: {
          url: 'https://i.ytimg.com/vi/SiDYFCd9SH4/hqdefault.jpg',
          width: 480,
          height: 360,
        },
      },
      channelTitle: 'Fox News',
      liveBroadcastContent: 'none',
      publishTime: '2020-09-26T02:26:02Z',
    },
  },
  {
    kind: 'youtube#searchResult',
    etag: 'rk64pdHo_kvd3E4eYbGjC5LDg6Y',
    id: {
      kind: 'youtube#video',
      videoId: 'WRoZgfjzGNQ',
    },
    snippet: {
      publishedAt: '2020-09-20T04:00:00Z',
      channelId: 'UCLkAepWjdylmXSltofFvsYQ',
      title: 'BTS (방탄소년단) &#39;Dynamite&#39; (&#39;70s remix) MV',
      description:
        'BTS #방탄소년단 #BTS_Dynamite Connect with BTS: https://ibighit.com/bts http://twitter.com/BTS_bighit http://twitter.com/BTS_twt ...',
      thumbnails: {
        default: {
          url: 'https://i.ytimg.com/vi/WRoZgfjzGNQ/default.jpg',
          width: 120,
          height: 90,
        },
        medium: {
          url: 'https://i.ytimg.com/vi/WRoZgfjzGNQ/mqdefault.jpg',
          width: 320,
          height: 180,
        },
        high: {
          url: 'https://i.ytimg.com/vi/WRoZgfjzGNQ/hqdefault.jpg',
          width: 480,
          height: 360,
        },
      },
      channelTitle: 'BANGTANTV',
      liveBroadcastContent: 'none',
      publishTime: '2020-09-20T04:00:00Z',
    },
  },
  {
    kind: 'youtube#searchResult',
    etag: 'Oui4WDt1DiTfEnnFwmIYvXddG48',
    id: {
      kind: 'youtube#video',
      videoId: 'ZujP5qjviLc',
    },
    snippet: {
      publishedAt: '2020-09-18T14:00:02Z',
      channelId: 'UCXooz9whNJZBRTHi9AqdjPw',
      title: 'Gusttavo Lima - De Menina Pra Mulher (O Embaixador The Legacy)',
      description:
        'Essa música faz parte do novo álbum do Gusttavo Lima, O Embaixador The Legacy, gravado em Goiânia/GO. INSCREVA-SE NO CANAL: ...',
      thumbnails: {
        default: {
          url: 'https://i.ytimg.com/vi/ZujP5qjviLc/default.jpg',
          width: 120,
          height: 90,
        },
        medium: {
          url: 'https://i.ytimg.com/vi/ZujP5qjviLc/mqdefault.jpg',
          width: 320,
          height: 180,
        },
        high: {
          url: 'https://i.ytimg.com/vi/ZujP5qjviLc/hqdefault.jpg',
          width: 480,
          height: 360,
        },
      },
      channelTitle: 'Gusttavo Lima Oficial',
      liveBroadcastContent: 'none',
      publishTime: '2020-09-18T14:00:02Z',
    },
  },
  {
    kind: 'youtube#searchResult',
    etag: 'l7gYGxv9Fe1QVWxe9kmOCGuhCeU',
    id: {
      kind: 'youtube#video',
      videoId: 'E8Im8orjfho',
    },
    snippet: {
      publishedAt: '2020-09-24T19:45:00Z',
      channelId: 'UCaXkIU1QidjPwiAYu6GcHjg',
      title: 'Watch All In With Chris Hayes Highlights: September 23 | MSNBC',
      description:
        'Get the latest news and commentary from Chris Hayes weekdays at 8 p.m. ET on MSNBC. » Subscribe to MSNBC: http://on.msnbc.com/SubscribeTomsnbc ...',
      thumbnails: {
        default: {
          url: 'https://i.ytimg.com/vi/E8Im8orjfho/default.jpg',
          width: 120,
          height: 90,
        },
        medium: {
          url: 'https://i.ytimg.com/vi/E8Im8orjfho/mqdefault.jpg',
          width: 320,
          height: 180,
        },
        high: {
          url: 'https://i.ytimg.com/vi/E8Im8orjfho/hqdefault.jpg',
          width: 480,
          height: 360,
        },
      },
      channelTitle: 'MSNBC',
      liveBroadcastContent: 'none',
      publishTime: '2020-09-24T19:45:00Z',
    },
  },
  {
    kind: 'youtube#searchResult',
    etag: 'JrIOfn8SO_0vifjfOfpPtSpYsZE',
    id: {
      kind: 'youtube#video',
      videoId: 'IyqPvdQvEIc',
    },
    snippet: {
      publishedAt: '2020-09-18T19:42:39Z',
      channelId: 'UCVeT85DCQZ4urbnmce1k0BQ',
      title: 'Juriul X Factor, în lacrimi! Super 4 cântă Caruso - Lucio Dalla',
      description:
        'Super 4 au cântat la cererea juraților! |Lucio Dalla - Caruso @The X Factor Romania, în fiecare vineri, de la 20:30, la Antena 1! #XFactor #Antena1 #deneoprit ...',
      thumbnails: {
        default: {
          url: 'https://i.ytimg.com/vi/IyqPvdQvEIc/default.jpg',
          width: 120,
          height: 90,
        },
        medium: {
          url: 'https://i.ytimg.com/vi/IyqPvdQvEIc/mqdefault.jpg',
          width: 320,
          height: 180,
        },
        high: {
          url: 'https://i.ytimg.com/vi/IyqPvdQvEIc/hqdefault.jpg',
          width: 480,
          height: 360,
        },
      },
      channelTitle: 'The X Factor Romania',
      liveBroadcastContent: 'none',
      publishTime: '2020-09-18T19:42:39Z',
    },
  },
];

export default videos;
