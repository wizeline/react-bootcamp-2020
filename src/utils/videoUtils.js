const flattenVideoStructure = (video) => {
  if (!video) return;
  return {
    ...video.id,
    ...video.snippet,
  };
};

export { flattenVideoStructure };
