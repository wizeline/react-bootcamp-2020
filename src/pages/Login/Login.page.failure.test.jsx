import React from 'react';
import { act } from '@testing-library/react';
import { render, screen } from '../../utils/test-utils'
import { BrowserRouter } from 'react-router-dom';
import userEvent from '@testing-library/user-event';
import LoginPage from './Login.page';
import Adapter from 'enzyme-adapter-react-16';
import { configure } from 'enzyme';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

configure({adapter: new Adapter()});
const mockStore = configureStore([]);

describe('LoginPage failure scenario', () => {
  let store;
  let component;
 
  beforeEach(() => {
    store = mockStore({
      myState: 'sample text',
    });
    store.dispatch = jest.fn();
    component = (
      <Provider store={store}>
        <BrowserRouter><LoginPage /></BrowserRouter>
      </Provider>
    );
  });

  it('writes inside the fields and click button ', async () => {
    act(() => {
      render(component, {
        initialState: {
          auth: {},
          videos: {}
        }
      })
    });
    
    await act(async() => {
      const usernameInput = screen.getByTestId('username-field');
      const passwordInput = screen.getByTestId('password-field');
      const loginButton = screen.getByTestId('login-submit-button');
      await userEvent.type(usernameInput, 'wizeline');
      await userEvent.type(passwordInput, 'Rocks!');
      expect(usernameInput.value).toEqual('wizeline');
      expect(passwordInput.value).toEqual('Rocks!');
      userEvent.click(loginButton, 'login-submit-button')
    });
  });
});