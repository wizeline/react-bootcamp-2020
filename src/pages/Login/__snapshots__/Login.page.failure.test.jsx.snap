// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Connected React-Redux Component should render with given state from Redux store 1`] = `
<section
  className="login"
>
  <h1>
    Welcome to Videos!
  </h1>
  <form
    className="login-form"
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <label
        htmlFor="username"
      >
        <strong>
          username 
        </strong>
        <input
          data-testid="username-field"
          id="username"
          onChange={[Function]}
          required={true}
          type="text"
        />
      </label>
    </div>
    <div
      className="form-group"
    >
      <label
        htmlFor="password"
      >
        <strong>
          password 
        </strong>
        <input
          data-testid="password-field"
          id="password"
          onChange={[Function]}
          required={true}
          type="password"
        />
      </label>
    </div>
    <button
      data-testid="login-submit-button"
      type="submit"
    >
      login
    </button>
  </form>
</section>
`;
